src/utils/dateParser.ts

```diff
@@ -0,0 +1,13 @@
+import { isValid, parse } from 'date-fns';
+
+const parser = (value: string): Date | null => {
```

Comment:
Rename file and handler to `parseDate`. Param should be named to `datestring`. Also you don't need to tell explicitly the type here and in the `validateIfFutureDate`